src/REPLMode/REPLMode.jl

```diff
@@ -400,9 +398,22 @@ function do_cmd(repl::REPL.AbstractREPL, input::String; do_rethrow=false)
     end
 end
 
-function do_cmd!(command::Command, repl)
+
+function do_cmds(commands::Vector{Command}, io)
+    if !isinteractive() && !TEST_MODE[] && !PRINTED_REPL_WARNING[]
+        @warn "The Pkg REPL mode is intended for interactive use only, and should not be used from scripts. It is recommended to use the functional API instead."
+        PRINTED_REPL_WARNING[] = true
+    end
```

Comment:
maxlog=1 instead?

Reply:
Probably, but maybe that can be another PR to keep this part a pure code movement.

Reply:
Ah yeah ok